tools/OpenConsole.psm1

```diff
@@ -74,16 +74,15 @@ function Set-MsbuildDevEnvironment
         default { throw "Unknown architecture: $switch" }
     }
 
-    $vcvarsall = "$vspath\VC\Auxiliary\Build\vcvarsall.bat"
+    $devShellModule = "$vspath\Common7\Tools\Microsoft.VisualStudio.DevShell.dll"
+
+    Import-Module -Global -Name $devShellModule
 
     Write-Verbose 'Setting up environment variables'
-    cmd /c ("`"$vcvarsall`" $arch & set") | ForEach-Object {
-        if ($_ -match '=')
-        {
-            $s = $_.Split("=");
-            Set-Item -force -path "env:\$($s[0])" -value "$($s[1])"
-        }
-    }
+    Enter-VsDevShell -VsInstallPath $vspath -SkipAutomaticLocation `
+        -devCmdArguments "-arch=$arch" | Out-Null
+        
+    Set-Item -Force -path "Env:\Platform" -Value $arch | Out-Null

```

Comment:
This is much much better than "parse the environment variables directly"! Do you know how portable it is? Does this work on both PSCore and PS5?

Reply:
yes it does work on ps 5 and 7, didn't check on 6.
this might not work if the version of visual studio installed doesn't have developer PowerShell, but i think all versions of vs 2019 have it, so not that big of a deal